numpy/ma/tests/test_core.py

```diff
@@ -1351,16 +1351,24 @@ def test_minmax_dtypes(self):
             assert masked_array([-cmax, 0], mask=[0, 1]).max() == -cmax
             assert masked_array([cmax, 0], mask=[0, 1]).min() == cmax
 
-    @pytest.mark.parametrize("time_type", [np.datetime64("NaT", 's'),
-                                          np.timedelta64("NaT", 's')])
+    @pytest.mark.parametrize("time_type", ["M8[s]", "m8[s]"])
     def test_minmax_time_dtypes(self, time_type):
-        # Additional tests on max/min for time dtypes
-        x = np.array([1, 1, -2, 4, 5, -10, 10, 1, 2, 3], dtype= time_type)
-        m1 = [1, 0, 0, 0, 0, 0, 1, 0, 0, 0]
-        xm = masked_array(x, mask= m1)
+        def minmax_with_mask(arr, mask):
+            masked_arr = masked_array(arr, mask=mask)
+            expected_min = arr[~np.array(mask, dtype=bool)].min()
+            expected_max = arr[~np.array(mask, dtype=bool)].max()
 
-        assert_equal(np.int64(xm.min()), -10)
-        assert_equal(np.int64(xm.max()), 5)
+            assert_equal(np.int64(masked_arr.min()), np.int64(expected_min))
+            assert_equal(np.int64(masked_arr.max()), np.int64(expected_max))
+        # Additional tests on max/min for time dtypes
+        x1 = np.array([1, 1, -2, 4, 5, -10, 10, 1, 2, 3], dtype=time_type)
+        x2 = np.array(['NaT', 1, -2, 4, 5, -10, 10, 1, 2, 3], dtype=time_type)
+        x3 = np.array(['NaT', 'NaT', -2, 4, 5, -10, 10, 1, 2, 3], dtype=time_type)
+        x_test = [x1, x2, x3]
```

Comment:
Might have been nice with a parametrization, but this is good.  Thanks for adding a NaT test.